### PR TITLE
Adding more animations and overheating effect to the sensors suite

### PR DESCRIPTION
**Pull request by Sindorman**
For fun decided to animate sensor suite based off it's range setting and add special effect for when the heat exceeds 85%.
dreamseeker_GDjnQ2EN5z

### CODE REVIEW

**Comment by Sindorman** [comment]
!review

**Review by SleepyGemmy — APPROVED** [review]

**Review by alsoandanswer — APPROVED** [review]

File: code/modules/overmap/ships/computers/sensors.dm [hunk 1]
```diff
@@ -237,10 +237,35 @@
...
-	if(use_power)
-		icon_state = "sensors"
+	icon_state = "sensors_off"
+	if(!use_power)
+		cut_overlays()
+		return
+
+	var/overlay = "sensors-effect"
+
+	var/range_precentage = range / world.view * 100
```
Comment (alsoandanswer) [inline code/modules/overmap/ships/computers/sensors.dm:247]:
```suggestion
	var/range_percentage = range / world.view * 100
```

File: code/modules/overmap/ships/computers/sensors.dm [hunk 2]
```diff
@@ -237,10 +237,35 @@
...
+
+	if(range_precentage < 20)
```
Comment (alsoandanswer) [inline code/modules/overmap/ships/computers/sensors.dm:249]: can this be a switch? if not then carry on

File: code/modules/overmap/ships/computers/sensors.dm [hunk 3]
```diff
@@ -237,10 +237,35 @@
...
+	else if(range_precentage < 60)
+		overlay = "[overlay]3"
+	else if(range_precentage < 80)
+		overlay = "[overlay]4"
 	else
-		icon_state = "sensors_off"
+		overlay = "[overlay]5"
+
+	// Check if we are already using this overlay. Since updating is expensive.
+	if (!(overlay in our_overlays))
```
Comment (SleepyGemmy) [inline code/modules/overmap/ships/computers/sensors.dm:261]:
```suggestion
	if(!(overlay in our_overlays))
```

**Comment by SleepyGemmy** [comment]
see alsoandanswer's comment.

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]
see above comments.

File: code/modules/overmap/ships/computers/sensors.dm (hunk 2 above)
Comment (Sindorman) [reply code/modules/overmap/ships/computers/sensors.dm:249]: Cannot be a switch statement.

**Comment by Sindorman** [comment]
!review

File: code/modules/overmap/ships/computers/sensors.dm [hunk 4]
```diff
@@ -237,10 +237,35 @@
...
 	else
-		icon_state = "sensors_off"
+		overlay = "[overlay]5"
+
+	// Check if we are already using this overlay. Since updating is expensive.
+	if(!(overlay in our_overlays))
+		cut_overlays()
+		add_overlay(overlay)
+
+		var/heat_precentage = heat / critical_heat * 100
```
Comment (SleepyGemmy) [inline code/modules/overmap/ships/computers/sensors.dm:265]: heat percentage is still misspelled.

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**